### PR TITLE
Use Go modules (replaces dep)

### PR DESCRIPTION
**Pull request by hwh33**
Tests are failing for me in the same manner in which they fail on master.  The binary builds without error.

### CODE REVIEW

**Comment by hwh33** [comment]
Readme diff got mangled.  Probably easier to just look at the [new version](https://github.com/getlantern/http-proxy/blob/harry/use-go-modules/README.md).  I added a new section at the beginning about using modules.  I also simplified the "Run" section since we no longer need to instruct users to pull down dependencies (`go run` will do that automatically when using modules).

**Comment by hwh33** [comment]
Also looks like the tests run on master okay, but not this branch.  Looking into it.  Don't merge this for now.

File: server/server_test.go [hunk 1]
```diff
@@ -382,7 +382,7 @@ func TestDirectOK(t *testing.T) {
 		}
 
 		resp, _ := http.ReadResponse(bufio.NewReader(conn), nil)
-		assert.Equal(t, http.StatusBadGateway, resp.StatusCode, "should fail")
+		assert.Equal(t, http.StatusBadRequest, resp.StatusCode, "should fail")
```
Comment (hwh33) [inline server/server_test.go]: Okay, I figured out the failing tests.  It was actually due to a change in the behavior of the http package introduced in Go 1.12.  I'm not sure if this is the appropriate fix - do we want to return Bad Gateway specifically or do we just want to return any sensible error status?

This is orthogonal to the Go mod changes and is instead just related to Go 1.12 support.  If we need to do something more complicated than this drive-by fix, maybe I should separate this into its own PR.

Comment (hwh33) [reply server/server_test.go]: Here's the relevant part of the Go 1.12 changelog: https://golang.org/doc/go1.12#net/http

**Comment by bcmertz** [comment]
I'm seeing similar additions of ```// indirect ``` being added to go.mod and a couple extra dependencies being added to go.sum

File: server/server_test.go (hunk 1 above)
Comment (bcmertz) [reply server/server_test.go]: Good catch! It seems to me that this change is semantically more in line with what we were intending to check - 	
```// HTTPS can't be tunneled using Direct Proxying, as redirections
// require a TLS handshake between the proxy and the origin```
It seems a 400 BadRequest error code would be more appropriate to check for than 502 BadGateway in this case.

**Comment by hwh33** [comment]
Okay, I think the changes to go.mod and go.sum were because I didn't run `go mod tidy`.  [More explanation here](https://github.com/getlantern/ddftool/pull/3#issuecomment-504555052).  Should be fixed now.  Thanks again for catching that!

File: server/server_test.go (hunk 1 above)
Comment (hwh33) [reply server/server_test.go]: Agreed!

**Comment by bcmertz** [comment]
LGTM!

**Comment by hwh33** [comment]
Thanks @bcmertz!

**Pull request merged**